### ISSUE TITLE
BaseLLM issue with ContextRelevancy and AnswerRelevancy

### ISSUE THREAD

**Issue by arikupe** [explodinggradients/ragas#110]
Hello! Thanks for the great work :) 

I've run into a bug while trying to use a local LLM. I cannot compute either `ContextRelevancy` or `AnswerRelevancy` when using a langchain BaseLLM, due to this Exception in line 36 in `ragas.metrics.llms`:

```
20      def generate(
21          prompts: list[ChatPromptTemplate],
22          llm: BaseLLM | BaseChatModel,
23          n: t.Optional[int] = None,
24          temperature: float = 0,
25          callbacks: t.Optional[Callbacks] = None,
26      ) -> LLMResult:
27          old_n = None
28          n_swapped = False
29          llm.temperature = temperature
30          if n is not None:
31               if isinstance(llm, OpenAI) or isinstance(llm, ChatOpenAI):
32                    old_n = llm.n
33                    llm.n = n
34                    n_swapped = True
35               else:
36  --->           raise Exception(
37                     f"n={n} was passed to generate but the LLM {llm} does not support it."
38                     " Raise an issue if you want support for {llm}."
39                )
```

The issue arises because when `ContextRelevancy` and `AnswerRelevancy` call this function they pass in `n=self.strictness`, eg in `ragas.metrics.answer_relevance`:

```
75        results = generate(
76            prompts,
77            self.llm,
78            n=self.strictness,
79            temperature=self.temperature,
80            callbacks=batch_group,
81        )
```

However `strictness` must be an integer due to both class's post_init:
```
54    def __post_init__(self: t.Self):
55        self.temperature = 0.2 if self.strictness > 0 else 0
```

Passing `strictness=None` would resolve the Exception issue but yields another error due to the post_init integer comparison.

Is there any scope to change the Exception to allow for n=0 or similar (eg change line 30 in `ragas.metrics.llms` to `if n is not None and n>0`?

If not, we are currently unable to use these metrics with a non-OpenAI LLM as far as I can tell.

Thank you!

**Comment by jjmachan** [comment]
Thank you for pointing this out @arikupe, sadly This is a current limitation of ragas ATM. I'm working on a fix and will merge it by next week. I'll reach out to you right after that fix has been released.

I hope you can wait till  the mean time

**Comment by arikupe** [comment]
Hi @jjmachan, thanks so much for looking into this!

Of course, I can use OpenAI for now. Thanks again :)

**Comment by venuraja79** [comment]
Thanks a lot for this great repo. I see the same issue with ContextRecall as well. This fix will be great to have whenever it is released.

**Comment by jjmachan** [comment]
this has been fixed with #197 and will be released shortly, please do try it out and reopen the issue if its still a problem :)
